### PR TITLE
Add unsign member endpoint

### PR DESCRIPTION
**Pull request by tobiasbexelius**
# Jira Issue: [MX-146] 

## What?
- An endpoint to unsign a previously signed member, available locally and in staging only.

## Why?
- To allow partners to sign multiple times with the same ssn for testing purposes 

## Optional checklist
- [x] Codescouted
- [ ] Unit tests written
- [x] Tested locally

[MX-146]: https://hedvig.atlassian.net/browse/MX-146

### CODE REVIEW

File: src/main/kotlin/com/hedvig/rapio/externalservices/memberService/MemberServiceClient.kt [hunk 1]
```diff
@@ -8,7 +8,7 @@ import org.springframework.web.bind.annotation.PostMapping
 import org.springframework.web.bind.annotation.RequestBody
 
 @FeignClient(
-    name = "payment-service",
+    name = "member-service",
```
Comment (vonElfvin) [inline src/main/kotlin/com/hedvig/rapio/externalservices/memberService/MemberServiceClient.kt:11]: 🕵️

File: src/main/kotlin/com/hedvig/rapio/qa/QaController.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+@Profile("staging", "development")
+@RequestMapping("v1/staging/")
+class QaController(
+    private val qaService: QaService
+) {
+    @PostMapping("unsignMember")
```
Comment (vonElfvin) [inline src/main/kotlin/com/hedvig/rapio/qa/QaController.kt:18]:
```suggestion
    @PostMapping("unsign-member")
```

File: src/main/kotlin/com/hedvig/rapio/qa/QaMemberServiceClient.kt [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import com.hedvig.rapio.externalservices.memberService.dto.IsMemberRequest
+import com.hedvig.rapio.qa.dto.MemberServiceUnsignMemberRequest
+import com.hedvig.rapio.qa.dto.UnsignMemberRequest
+import org.springframework.cloud.openfeign.FeignClient
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+
+@FeignClient(
+    name = "member-service",
```
Comment (vonElfvin) [inline src/main/kotlin/com/hedvig/rapio/qa/QaMemberServiceClient.kt:12]: Unsure how FeignClient works, but can we have two with the same name, and the same URL?
```suggestion
    name = "member-service",
```

File: src/main/kotlin/com/hedvig/rapio/qa/QaMemberServiceClient.kt [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    url = "\${hedvig.member-service.url:member-service}"
+)
+interface QaMemberServiceClient {
```
Comment (vonElfvin) [inline src/main/kotlin/com/hedvig/rapio/qa/QaMemberServiceClient.kt:15]: I'm not the biggest fan of "Qa" but I honestly don't know of a better name, this is just a bad suggestion
```suggestion
interface StagingToolsMemberServiceClient {
```

File: src/main/kotlin/com/hedvig/rapio/qa/QaController.kt [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    fun unsignMember(
+        @RequestBody request: UnsignMemberRequest
+    ): ResponseEntity<UnsignMemberResponse> = ResponseEntity.ok(
+        UnsignMemberResponse(
+            qaService.unsignMember(
+                personalNumber = request.personalNumber
+            )
+        )
+    )
```
Comment (vonElfvin) [inline src/main/kotlin/com/hedvig/rapio/qa/QaController.kt:27]: I personally like "noContent" if the result is not interesting. 

```suggestion
    ): ResponseEntity<Void> {
        val success = qaService.unsignMember(personalNumber = request.personalNumber)
        if (success) {
          return ResponseEntity.noContent().build()
        } else {
          return ResponseEntity.badGateway().build()
        }
    }
```

**Review by vonElfvin — APPROVED** [review]
This will do wonders :) Added some styling comments to consider if you like

File: src/main/kotlin/com/hedvig/rapio/qa/QaMemberServiceClient.kt (hunk 3 above)
Comment (tobiasbexelius) [reply src/main/kotlin/com/hedvig/rapio/qa/QaMemberServiceClient.kt:12]: Seems like it, I did not run into any problems when running locally

File: src/main/kotlin/com/hedvig/rapio/qa/QaController.kt (hunk 5 above)
Comment (tobiasbexelius) [reply src/main/kotlin/com/hedvig/rapio/qa/QaController.kt:27]: I like that, but the error should probably be notFound

File: src/main/kotlin/com/hedvig/rapio/qa/dto/UnsignMemberResponse.kt [hunk 6]
```diff
@@ -0,0 +1,5 @@
+package com.hedvig.rapio.qa.dto
+
+data class UnsignMemberResponse(
+    val isUnsigned:Boolean
```
Comment (vonElfvin) [inline src/main/kotlin/com/hedvig/rapio/qa/dto/UnsignMemberResponse.kt:4]: Format on save, or Option-Cmd-L
```suggestion
    val isUnsigned: Boolean
```

**Review by vonElfvin — APPROVED** [review]
Clean!

File: src/main/kotlin/com/hedvig/rapio/qa/dto/MemberServiceUnsignMemberRequest.kt [hunk 7]
```diff
@@ -0,0 +1,5 @@
+package com.hedvig.rapio.qa.dto
+
+data class MemberServiceUnsignMemberRequest(
+    val ssn: String
+)
```
Comment (Michael-Duivestein) [inline src/main/kotlin/com/hedvig/rapio/qa/dto/MemberServiceUnsignMemberRequest.kt:5]: Newline

File: src/main/kotlin/com/hedvig/rapio/qa/QualityAssuranceController.kt [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+import com.hedvig.rapio.qa.dto.UnsignMemberRequest
+import org.springframework.context.annotation.Profile
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+@Profile("staging", "development")
```
Comment (Michael-Duivestein) [inline src/main/kotlin/com/hedvig/rapio/qa/QualityAssuranceController.kt:12]: Does this mean these endpoints are for non-production purposes (things to be used while testing stuff)?

Comment (tobiasbexelius) [reply src/main/kotlin/com/hedvig/rapio/qa/QualityAssuranceController.kt:12]: Yes, they are for testing only

**Review by vonElfvin — APPROVED** [review]

**Pull request merged**